src/components/input/Input.js

```diff
@@ -0,0 +1,52 @@
+import classnames from 'classnames';
+
+import { downcastRef } from '../../util/typing';
+
+import { inputGroupStyles } from './InputGroup';
+
+/**
+ * @typedef {import('../../types').PresentationalProps} CommonProps
+ * @typedef {import('preact').JSX.HTMLAttributes<HTMLInputElement>} HTMLInputAttributes
+ *
+ * @typedef InputProps
+ * @prop {boolean} [hasError=false]
+ * @prop {'email'|'search'|'text'|'url'} [type="text"]
+ */
+
+/**
+ * Render a text field input
+ *
+ * @param {CommonProps & InputProps & Omit<HTMLInputAttributes, 'type'>} props
+ */
+const InputNext = function Input({
+  children,
+  classes,
+  elementRef,
+
+  hasError,
+  type = 'text',
+
+  ...htmlAttributes
+}) {
+  return (
+    <input
+      {...htmlAttributes}
+      type={type}
+      ref={downcastRef(elementRef)}
+      className={classnames(
+        'focus-visible-ring ring-inset border rounded-sm w-full p-2',
+        'bg-grey-0 focus:bg-white disabled:bg-grey-1',
+        'placeholder:text-color-grey-5 disabled:placeholder:color-grey-6',
```

Comment:
The disabled and non-disabled states are visually quite similar. I wonder if that could cause confusion. Looking around I found that GitHub uses a fairly similar style for some of its most common inputs (eg. comment box) and I've never noticed until now.

<img width="858" alt="Enabled vs disabled" src="https://user-images.githubusercontent.com/2458/182360469-39e74079-5363-4fc2-9065-923bbb1d749c.png">